cEP-0018.md

```diff
@@ -0,0 +1,164 @@
+# Integration of ANTLR into coala core
+
+| Metadata |                                               |
+| -------- | --------------------------------------------- |
+| cEP      | 18                                            |
+| Version  | 1.0                                           |
+| Title    | Integration of ANTLR into coala core          |
+| Authors  | Viresh Gupta <mailto:viresh16118@iiitd.ac.in> |
+| Status   | Proposed                                      |
+| Type     | Feature                                       |
+
+## Abstract
+
+This document describes how an API based on ANTLR will be constructed and
+maintained.
+
+## Introduction
+
+ANTLR provides parsers for various language grammars and thus we can provide
+an interface from within coala and make it available to bear writers so that
+they can write advanced linting bears. This will aim at supporting the more
+flexible visitor based method of AST traversal (as opposed to listener based
+mechanisms).
+
+The proposal is to introduce a parallel concept to the coala-bears library,
+which will be called the coala-ast hereon.
+
+## Proposed Change
+
+Here is the detailed implementation stepwise:
+
+1. There will be a separate repository named as coala-antlr-ast which will
+   be installable via `pip install coala-antlr-ast`.
+2. A new package would be introduced in the coala-antlr-ast repository,
+   which would be the maintain all the dependencies for antlr-ast based
+   bears. This would be the `antlrastlib` package
```

Comment:
if it is going to be coala specific, i.e. invoke coala libraries or link to coala specific APIs, then the package module name should have `coa` in it, and probably a `coala-` prefix based on https://github.com/coala/cEPs/issues/17

Reply:
How about coant ?